### PR TITLE
feat: change wishlist action buttons to the UI Kit dropdown menu

### PR DESCRIPTION
**Pull request by NaMax66**
## Description
Change the setting btns to the UI Kit component

### Jira-link:
https://virtocommerce.atlassian.net/browse/ST-5393
### Artifact URL:
https://vc3prerelease.blob.core.windows.net/packages/vc-theme-b2b-vue-1.44.0-pr-865-116c-116cf689.zip
https://vc3prerelease.blob.core.windows.net/packages/vc-theme-b2b-vue-1.44.0-pr-865-be08-be08f6db.zip

### CODE REVIEW

File: client-app/shared/wishlists/components/wishlist-dropdown-menu.vue [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+    color: "secondary",
+    localePath: "shared.wishlists.list_card.make_private_button",
+  },
+  [WishlistScopeType.Organization]: {
+    scope: WishlistScopeType.Organization,
+    icon: "users",
+    color: "info",
+    localePath: "shared.wishlists.list_card.share_with_organisation_button",
+  },
+};
```
Comment (asvishnyakov) [inline client-app/shared/wishlists/components/wishlist-dropdown-menu.vue:55]: Same as above

File: client-app/shared/wishlists/components/wishlist-dropdown-menu.vue [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+interface IEmit {
+  (event: "edit"): void;
+  (event: "remove"): void;
+  (event: "setScope", scope: WishlistScopeType): void;
+}
+
+interface IProps {
+  currentScope?: WishlistScopeType;
+  isCorporateMember: boolean;
+}
```
Comment (asvishnyakov) [inline client-app/shared/wishlists/components/wishlist-dropdown-menu.vue:66]: I believe there is eslint rule about these interfaces and `defineEmits, defineProps` (they should be the first in the file). Didn't it work?

**Review by asvishnyakov — CHANGES_REQUESTED** [review]

File: client-app/shared/wishlists/components/wishlist-dropdown-menu.vue [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+    </template>
+    <template #content>
+      <VcMenuItem color="secondary" @click="$emit('edit')">
+        <VcIcon name="pencil" />
+
+        <span>{{ $t("shared.wishlists.list_card.list_edit_button") }}</span>
+      </VcMenuItem>
+      <VcMenuItem
+        v-if="isCorporateMember && scopeButton"
+        :color="scopeButton.color"
```
Comment (goldenmaya) [inline client-app/shared/wishlists/components/wishlist-dropdown-menu.vue:14]: color="secondary"

File: client-app/shared/wishlists/components/wishlist-dropdown-menu.vue [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+        @click="$emit('setScope', scopeButton.scope)"
+      >
```
Comment (goldenmaya) [inline client-app/shared/wishlists/components/wishlist-dropdown-menu.vue:16]: use v-if Private v-else

File: client-app/shared/wishlists/components/wishlist-dropdown-menu.vue [hunk 5]
```diff
@@ -0,0 +1,77 @@
...
+        <VcIcon :name="scopeButton.icon" />
```
Comment (goldenmaya) [inline client-app/shared/wishlists/components/wishlist-dropdown-menu.vue:17]: use class="..." to change icon color
For example:
class="text-[--color-accent-600]"

**Review by goldenmaya — CHANGES_REQUESTED** [review]

File: client-app/shared/wishlists/components/wishlist-dropdown-menu.vue (hunk 2 above)
Comment (NaMax66) [reply client-app/shared/wishlists/components/wishlist-dropdown-menu.vue:66]: We have warning for `vue/define-props-declaration` rule
But do not have "interface always on top" kind of rule

File: client-app/pages/account/lists.vue [hunk 6]
```diff
@@ -126,28 +111,8 @@ function openDeleteListModal(list: WishlistType) {
...
-      title: t("common.buttons.settings"),
-      classes: "bg-[--color-neutral-500]",
-      clickHandler(list: WishlistType) {
-        openListSettingsModal(list);
-      },
-    },
-  ];
-
-  return actions;
+function setScope(listId: string, scope: WishlistScopeType) {
```
Comment (asvishnyakov) [inline client-app/pages/account/lists.vue:114]: Emits can be async, so make function async and await for wishlist update

**Review by asvishnyakov — APPROVED** [review]

**Review by goldenmaya — APPROVED** [review]

**Pull request merged**